js/application.js

```diff
@@ -6,8 +6,5 @@ timerLimitRender();
 renderSortingPanel();
 
 renderOnSiteCounterBar();
-addSortMoviesHandler();
 
-let currentMoviesList = listOfMovies;
-
-renderMovieCards(currentMoviesList);
\ No newline at end of file
+fetchAllMovies(renderMovieCards);
```

Comment:
Adding callbacks as arguments to your fetch function complicates your code. It also forces you to always make a new fetch before using any app feature (sorting, searching). This leads to a slower app and complicated code. So, I recommend you keep using the global variable `listOfMovies` and initialise it with the API data.

```js
let listOfMovies = [];

fetch('...')
  .then(response => response.json())
  .then(data => listOfMovies = data)
  .catch(...);
```

Reply:
@vdaguenet Not working code sample
see [here|https://github.com/halynabondar/Movie_app/blob/not_working_code/js/list.js]

Reply:
I checked the non working branch and your issue is that `application.js` is executed before the API call ended. There are multiple ways to fix it.

## Option 1: return a Promise from list.js
```js
// list.js
let listOfMovies = [];

function fetchMovies() {
    return fetch("https://raw.githubusercontent.com/halynabondar/halynabondar.github.io/main/Data/movieApp.json")
        .then((response) => {
            console.log('Response received!');
            return response.json();
        })
        .then(data => {
            console.log('Data received!');
            listOfMovies = data
        })
        .catch(error => {
            console.error('There has been a problem with your fetch operation:', error);
        });
}
```
```js
// application.js

let currentMoviesList = [];
fetchMovies().then(function() {
    currentMoviesList = listOfMovies;
    renderMovieCards(currentMoviesList);
});
```

## Option 2: make the API call from application.js
```js
// list.js
let listOfMovies = [];
```
```js
// application.js
let currentMoviesList = [];
fetch("https://raw.githubusercontent.com/halynabondar/halynabondar.github.io/main/Data/movieApp.json")
    .then((response) => {
        console.log('Response received!');
        return response.json();
    })
    .then(data => {
        console.log('Data received!');
        listOfMovies = data;
        currentMoviesList = listOfMovies;
        renderMovieCards(currentMoviesList);
    })
    .catch(error => {
        console.error('There has been a problem with your fetch operation:', error);
    });
```
If you choose this option, it would be good to rethink the usage of `listOfMovies` and `currentMoviesList`. You can probably refactor your app using only one of these two arrays.